samza-core/src/main/java/org/apache/samza/config/JobConfig.java

```diff
@@ -125,6 +134,16 @@
   public static final String CONTAINER_METADATA_FILENAME_FORMAT = "%s.metadata"; // Filename: <containerID>.metadata
   public static final String CONTAINER_METADATA_DIRECTORY_SYS_PROPERTY = "samza.log.dir";
 
+
+  // Auto-sizing related configs tthat ake precedence over respective sizing confings job.container.count, etc,
+  // *only* when job.autosizing.enabled is true. Otherwise current behavior is maintained.
+  public static final String JOB_AUTOSIZING_ENABLED = "job.autosizing.enabled";
+  public static final String JOB_AUTOSIZING_CONTAINER_COUNT = "job.autosizing.container.count";
+  public static final String JOB_AUTOSIZING_CONTAINER_THREAD_POOL_SIZE = "job.autosizing.container.thread.pool.size";
+  public static final String JOB_AUTOSIZING_CONTAINER_MAX_HEAP_MB = "job.autosizing.container.maxheap.mb";
+  public static final String JOB_AUTOSIZING_CONTAINER_MEMORY_MB = "job.autosizing.container.memory.mb";
+  public static final String JOB_AUTOSIZING_CONTAINER_MAX_CORES = "job.autosizing.container.cpu.cores";
+
```

Comment:
I think @rmatharu was working on merging some of this code in separately. You may want to double check with him on that. We should avoid merging the autosizing code with this PR, just to keep the changes isolated.

Reply:
Good point, this change is pulled in accidentally.